### ISSUE TITLE
Errors TC1 aren't raised with flakeheaven

### ISSUE THREAD

**Issue by ShaiAvr** [snok/flake8-type-checking#75]
I am using [flakeheaven](https://flakeheaven.readthedocs.io/en/latest/) (a maintained fork of the more known [flakehell](https://github.com/life4/flakehell) framework). I also use TC1 error ranges because I prefer postponed evaluation of annotations rather than string literals. However, I can't seem to generate any TC1 error. Take this code for example:
```python
from __future__ import annotations

from typing import TYPE_CHECKING

import numpy as np


if TYPE_CHECKING:
    from collections.abc import Iterable
    from typing import SupportsFloat
    import numpy.typing as npt


class Vector:
    _data: npt.NDArray[np.float64]

    def __init__(self, coordinates: Iterable[SupportsFloat | str]) -> None:
        ...

    @classmethod
    def of(cls) -> "Vector":
        pass
```
I would expect to get a `TC101` error telling me that I don't need a string literal for `Vector.of` return type. However, this code passes with no linter errors. On the other hand, this similar, but still wrong code:
```python
from typing import TYPE_CHECKING

import numpy as np


if TYPE_CHECKING:
    from collections.abc import Iterable
    from typing import SupportsFloat
    import numpy.typing as npt


class Vector:
    _data: npt.NDArray[np.float64]

    def __init__(self, coordinates: Iterable[SupportsFloat | str]) -> None:
        ...

    @classmethod
    def of(cls) -> Vector:
        pass
```
doesn't generate a `TC100` error telling me I am missing a `from __future__ import annotations` import. All I get is a regular flake8 `F821` error because of the "undefined" name `Vector`.

I tried enabling `TC1` range in the `pyproject.toml` file as described in the [documentation](https://flakeheaven.readthedocs.io/en/latest/config.html#plugins) (`select` and `enable-extensions` are [ignored](https://flakeheaven.readthedocs.io/en/latest/config.html#ignored-options) in flakeheaven), but nothing works.

Here are the relevant parts of the `pyproject.toml` file:
```toml
[tool.poetry]
version = "0.1.0"
description = ""
license = "MIT"
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.10"
numpy = "^1.22.2"

[tool.poetry.dev-dependencies]
ipython = "^8.1.1"
black = "^22.1.0"
mypy = "^0.931"
pre-commit = "^2.17.0"
isort = "^5.10.1"
flakeheaven = "^0.11.1"

# Flake8 plugins
flake8-type-checking = "^1.3.2"
flake8-bugbear = "^22.1.11"
flake8-builtins = "^1.5.3"
flake8-comprehensions = "^3.8.0"
flake8-simplify = "^0.18.1"

[tool.flakeheaven]
format = "grouped"
show_source = true
max_line_length = 88
max_complexity = 18
exclude = [".git", ".venv", "__pycache__", "docs/source/conf.py", "old", "build", "dist"]

[tool.flakeheaven.plugins]
"flake8-*" = ["+*"]  # Should be enough to work
# flake8-type-checking = ["+TC0*", "+TC1*"]  # Doesn't generate TC1 errors
flake8-type-checking = ["+*"]  # Doesn't generate TC1 errors
```

**Comment by sondrelg** [comment]
Yeah the error code opt-in/out in flake8 has always been a little hard to make sense, of to me at least. The way I tend to enable error code the plugin, is by using `enable-extensions=TC0,TC1` I think. 

Is there an equivalent of this in flakeheaven you could test? If that's doesn't work, perhaps we need to tweak the plugin somehow 🤔

I'm away for a few days, but can try to help more when back if 👆 doesn't help 😊

**Comment by ShaiAvr** [comment]
@sondrelg, As stated in their [documentation](https://flakeheaven.readthedocs.io/en/latest/config.html#plugins), the equivalent is to use `plugins`. However, I couldn't manage to make it work.

**Comment by sondrelg** [comment]
Looked at flakeheaven a little bit and I'm also not able to get it to work.

For a little bit of background: the way this plugin is intended to work is for the two error ranges TC1* and TC2* to be alternatives, which requires us to have them be opt-in. [This](https://github.com/snok/flake8-type-checking/blob/main/flake8_type_checking/plugin.py#L114) is where we do this. There are a few ways to opt-in; you can add the codes to the flake8 config's `select`, `extended_default_select`, or `enable_extensions` attributes. I think flake8-bugbear and a few other plugins do similar things for opt-in codes.

Do you know how flakeheaven enables error codes? The problem might be that they use another config option to enable codes/extensions.

Another explanation could be that the plugin error code declaration isn't quite right in this plugin, so flakeheaven isn't able to infer which codes should be selected. This seems to suggest this at least:

```
❯ flakeheaven codes flake8-type-checking
no codes found
```

**Comment by sondrelg** [comment]
I'm going to close this from inactivity for now. Feel free to reopen 🙂
